backends/dnf/pk-backend-dnf.c

```diff
@@ -1841,6 +1841,197 @@ dnf_utils_find_package_ids (DnfSack *sack, gchar **package_ids, GError **error)
 	return hash;
 }
 
+static void
+dnf_utils_process_dependency (DnfSack *sack,
+			      DnfPackage *pkg,
+			      PkRoleEnum role,
+			      gboolean recursive,
+			      GHashTable *visited,
+			      GPtrArray *results_array,
+			      GError **error)
+{
+	g_autoptr(GQueue) queue = g_queue_new ();
+	g_queue_push_tail (queue, g_object_ref (pkg));
+
+	/* mark the start package as visited so we don't process it again */
+	if (!g_hash_table_contains (visited, dnf_package_get_package_id (pkg)))
+		g_hash_table_add (visited, g_strdup (dnf_package_get_package_id (pkg)));
+
+	while (!g_queue_is_empty (queue)) {
+		g_autoptr(DnfPackage) curr = g_queue_pop_head (queue);
+		g_autoptr(DnfReldepList) reldeps = NULL;
+		guint n;
+
+		if (role == PK_ROLE_ENUM_DEPENDS_ON)
+			reldeps = dnf_package_get_requires (curr);
+		else
+			reldeps = dnf_package_get_provides (curr);
```

Comment:
Is this correct? "Provides" in many package-managers means "packages/dependencies provided by package X" and not "packages that require package X to function" (which is what `REQUIRED_BY` is).

This might be okay in DNF semantics, so I am just asking to double-check (same below).

Reply:
It's okay because further down we invert the test for the query to find things that either provide the requires or require the provides. :)